controllers/provider-alicloud/pkg/controller/infrastructure/add.go

```diff
@@ -32,13 +34,17 @@ type AddOptions struct {
 	Controller controller.Options
 	// IgnoreOperationAnnotation specifies whether to ignore the operation annotation or not.
 	IgnoreOperationAnnotation bool
+	// MachineImages is the default list of machine images.
+	MachineImages []config.MachineImage
+	// MachineImageOwnerSecretRef is the secret reference which contains credential of AliCloud subaccount for customized images.
+	MachineImageOwnerSecretRef corev1.SecretReference
```

Comment:
As MachineImageOwnerSecretRef could be a nil, please define it as *corev1.SecretReference